test/tests/jobs/GetJobTest.cpp

```diff
@@ -699,5 +700,100 @@ struct GetJobTest : tpunit::TestFixture {
+        string childJobID = tester->executeWaitVerifyContentTable(createJobCommand)["jobID"];
+        string childState = tester->readDB("SELECT state FROM jobs WHERE jobID=" + SQ(childJobID) + ";");
+        ASSERT_EQUAL(childState, "PAUSED");
+
+        // Wait a bit for retryAfter to kick in
+        sleep(3);
+
+        // The child job is still not ready.
+        getJobCommand["name"] = "ChildJob";
+        tester->executeWaitVerifyContent(getJobCommand, "404 No job found");
+
+        // Verify we get this job again, and everything is still in the expected states
+        getJobCommand["name"] = "ParentJob";
+        getJobResponse = tester->executeWaitVerifyContentTable(getJobCommand);
+        ASSERT_EQUAL(getJobResponse["jobID"], parentJobID);
+        state = tester->readDB("SELECT state FROM jobs WHERE jobID=" + SQ(parentJobID) + ";");
+        ASSERT_EQUAL(state, "RUNQUEUED");
+
+        childState = tester->readDB("SELECT state FROM jobs WHERE jobID=" + SQ(childJobID) + ";");
+        ASSERT_EQUAL(childState, "PAUSED");
+
+        // The child job is still not ready.
+        getJobCommand["name"] = "ChildJob";
+        tester->executeWaitVerifyContent(getJobCommand, "404 No job found");
+
+        // Finish the parent job this time
+        SData finishJobCommand("FinishJob");
+        finishJobCommand["jobID"] = parentJobID;
+        tester->executeWaitVerifyContentTable(finishJobCommand);
+
+        // Now the parent is PAUSED and the child is QUEUED for running.
+        state = tester->readDB("SELECT state FROM jobs WHERE jobID=" + SQ(parentJobID) + ";");
+        ASSERT_EQUAL(state, "PAUSED");
+        childState = tester->readDB("SELECT state FROM jobs WHERE jobID=" + SQ(childJobID) + ";");
+        ASSERT_EQUAL(childState, "QUEUED");
+
+        // The child job is ready.
+        getJobCommand["name"] = "ChildJob";
+        getJobResponse = tester->executeWaitVerifyContentTable(getJobCommand);
+        ASSERT_EQUAL(getJobResponse["jobID"], childJobID);
+
+        state = tester->readDB("SELECT state FROM jobs WHERE jobID=" + SQ(parentJobID) + ";");
+        ASSERT_EQUAL(state, "PAUSED");
+        childState = tester->readDB("SELECT state FROM jobs WHERE jobID=" + SQ(childJobID) + ";");
+        ASSERT_EQUAL(childState, "RUNNING");
+
+        // Finally finish the child so the parent can resume
+        finishJobCommand["jobID"] = childJobID;
+        tester->executeWaitVerifyContentTable(finishJobCommand);
+        childState = tester->readDB("SELECT state FROM jobs WHERE jobID=" + SQ(childJobID) + ";");
+        ASSERT_EQUAL(childState, "FINISHED");
+        state = tester->readDB("SELECT state FROM jobs WHERE jobID=" + SQ(parentJobID) + ";");
+        ASSERT_EQUAL(state, "QUEUED");
+        state = tester->readDB("SELECT nextRun FROM jobs WHERE jobID=" + SQ(parentJobID) + ";");
+
+        // Because we set nextRun in GetJob, we have to wait for 'retryAfter' time until the job is ready to be picked up again.
```

Comment:
This is the weirdness that took me a while to figure out...

I don't know if this is an actual issue or if this is fine. I think it's fine for the most part. I've looked, and I don't think it's super easy to fix this, but I can look further if you think I should.

Reply:
Why do you say it is weird? That's how it is designed, someone got it and we wait for it to finish processing it for the retryAfter time, if he didn't then we return it again so that someone else can process it.

Reply:
Oh wait... so after we finish correctly we don't re-run the parent till retryAfter has passed? If so it is a bug... not a big one, but a bug and I think we should fix it.

Reply:
Yes

Reply:
I just pushed a fix for that. Let me know what you think. If you think it's good, I'll fix the rest of your comments. Thanks!